diff --git a/src/rcw.c b/src/rcw.c
index ec78a0ab0..e01797496 100644
--- a/src/rcw.c
+++ b/src/rcw.c
@@ -2979,6 +2979,9 @@ static void rco_update_toolbar(RemminaConnectionObject *cnnobj)
 
 	viewonly_avail = remmina_protocol_widget_query_feature_by_type(REMMINA_PROTOCOL_WIDGET(cnnobj->proto), REMMINA_PROTOCOL_FEATURE_TYPE_VIEWONLY);
 	gtk_widget_set_sensitive(GTK_WIDGET(priv->toolitem_viewonly), viewonly_avail && cnnobj->connected);
+	if (viewonly_avail && cnnobj->connected) {
+		gtk_toggle_tool_button_set_active(GTK_TOGGLE_TOOL_BUTTON(cnnobj->cnnwin->priv->toolitem_viewonly), remmina_file_get_int(cnnobj->remmina_file, "viewonly", 0)==0? FALSE: TRUE);
+	}
 
 	switch (scalemode) {
 	case REMMINA_PROTOCOL_WIDGET_SCALE_MODE_NONE:
